### PR TITLE
Add docs for multi-dc support

### PR DESCRIPTION
**Pull request by BenWhitehead**

### CODE REVIEW

**Comment by BenWhitehead** [comment]
@ssk2 or @joel-hamill can I get some feedback on this new document?

**Comment by ssk2** [comment]
I've created an internal JIRA (DOCS-118) for @mattpennathe3rd and @joel-hamill.

File: docs/docs/multi-dc.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
+Multi Datacenter
+================
+
+As of version 0.2.0 Cassandra on Mesos allows for providing configuration
+to allow multiple datacenters to be aware of one another.
+
+
+## How it works
+
+Cassandra on Mesos allows an operator to configure the framework to provide the Datacenter and rack level information
```
Comment (ssk2) [inline docs/docs/multi-dc.md]: `Datacenter` -> `datacenter`

File: docs/docs/multi-dc.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+to Cassandra upon starting each server.
+
+Cassandra on Mesos can be configured to query other instances of Cassandra on Mesos to provide seed sets across
+multiple clusters in order to allow each of the clusters to coordinate and replicate data.  Periodically the seeds
+resource in the http API will be queried for each datacenter, when the seeds are able to be resolved they will be appended
```
Comment (ssk2) [inline docs/docs/multi-dc.md]: `http` -> `HTTP`

File: docs/docs/multi-dc.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+1. Ensure there are TCP routes allowing for traffic flow between each of the Mesos clusters.
+2. Start each cluster with the necessary configuration.
+3. Wait for each cluster to allocate it's seed nodes.
+4. After the seeds have been allocated a restart needs to be issued to all of the nodes that have already been started. Send a `POST` to the http API at `/cluster/rolling-restart/start` and the framework will restart each of the nodes so that new new seed set will take effect.
+5. Your clusters should now be able to replicate data between each other.
+
+## Example on DCOS
+
+Imagine three Data Centers dcos0, dcos1, dcos2 running three different DCOS Clusters.
```
Comment (ssk2) [inline docs/docs/multi-dc.md]: `Data Centers` -> `datacenters`

Comment (ssk2) [inline docs/docs/multi-dc.md]: `Clusters` -> `clusters`

File: docs/docs/multi-dc.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+The following options files would be used for the install for each instance of Cassandra on DCOS
```
Comment (ssk2) [inline docs/docs/multi-dc.md]: `The following options files can be used to install each instance of Cassandra on DCOS.`

File: docs/docs/multi-dc.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+          "name": "dcos1",
+          "url": "http://dcos1/service/cassandra.multi-dc"
+        }
+      ]
+    }
+  }
+}
+```
+
+1. On DCOS Cluster 0 install with `dcos package install cassandra --options=dcos0.json`
```
Comment (ssk2) [inline docs/docs/multi-dc.md]: Can we place these JSON files inline with the instructions?

File: docs/docs/multi-dc.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+2. On DCOS Cluster 1 install with `dcos package install cassandra --options=dcos1.json`
+3. On DCOS Cluster 2 install with `dcos package install cassandra --options=dcos2.json`
+4. Wait for each cluster to allocate it's seed nodes.
```
Comment (ssk2) [inline docs/docs/multi-dc.md]: `it's` -> `its`

File: docs/docs/multi-dc.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+5. After the seeds have been allocated a restart needs to be issued to all of the nodes that have already been started. Send a `POST` to the http API at `/cluster/rolling-restart/start` and the framework will restart each of the nodes so that new new seed set will take effect.
```
Comment (ssk2) [inline docs/docs/multi-dc.md]: `a restart` -> `a restart command`

File: docs/docs/multi-dc.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
+Multi Datacenter
+================
+
+As of version 0.2.0 Cassandra on Mesos allows for providing configuration
+to allow multiple datacenters to be aware of one another.
+
+
+## How it works
```
Comment (ssk2) [inline docs/docs/multi-dc.md]: It'd be really useful to add a high level explanation of why multiple datacenter support is useful/interesting. (You could also link to: http://www.datastax.com/dev/blog/deploying-cassandra-across-multiple-data-centers)

File: docs/docs/multi-dc.md (hunk 1 above)
Comment (joel-hamill) [inline docs/docs/multi-dc.md]: ...information. <-- add period

**Comment by BenWhitehead** [comment]
Updates incorporated

File: docs/docs/multi-dc.md (hunk 8 above)
Comment (joel-hamill) [inline docs/docs/multi-dc.md]: I'd reword to: "As of Cassandra 0.2.0, you can configure Cassandra on Mesos so that multiple datacenters are aware of each other."

File: docs/docs/multi-dc.md [hunk 9]
```diff
@@ -0,0 +1,180 @@
+---
+title: Multi Datacenter
+---
+
+# Multi Datacenter
+
+As of version 0.2.0 Cassandra on Mesos allows for providing configuration
```
Comment (ssk2) [inline docs/docs/multi-dc.md]: This sentence is hard to parse.

File: docs/docs/multi-dc.md [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+to allow multiple datacenters to be aware of one another.
+
+
+## How it works
+
+Cassandra on Mesos allows an operator to configure the framework to provide the Datacenter and rack level information
+to Cassandra upon starting each server.
+
+Cassandra on Mesos can be configured to query other instances of Cassandra on Mesos to provide seed sets across
```
Comment (joel-hamill) [inline docs/docs/multi-dc.md]: Change to: "Cassandra on Mesos can be configured to query other instances of itself to provide seed sets across multiple clusters so that each of the clusters can coordinate and replicate data."

File: docs/docs/multi-dc.md [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+multiple clusters in order to allow each of the clusters to coordinate and replicate data.  Periodically the seeds
+resource in the http API will be queried for each datacenter, when the seeds are able to be resolved they will be appended
+to the seed set when each server iss started.
```
Comment (joel-hamill) [inline docs/docs/multi-dc.md]: Typo: "...server is started."

File: docs/docs/multi-dc.md [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+
+
+## Requirements
+
+1. Each individual cluster will need to be able to route to the IP addresses of each of the other clusters.
```
Comment (joel-hamill) [inline docs/docs/multi-dc.md]: Change to: "Each individual cluster must be able to reach the IP addresses..."

File: docs/docs/multi-dc.md [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+2. All clusters need to have the same cluster name
```
Comment (joel-hamill) [inline docs/docs/multi-dc.md]: Change to: "All clusters must have the same cluster name."

File: docs/docs/multi-dc.md [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+3. Datacenters need to be defined before starting the framework. New datacenters can not yet be added after a cluster has initialized.
```
Comment (joel-hamill) [inline docs/docs/multi-dc.md]: Change to: "Datacenters must be defined before starting the framework. New datacenters cannot be added until after a cluster has initialized."

File: docs/docs/multi-dc.md [hunk 15]
```diff
@@ -0,0 +1,176 @@
...
+4. Node sizes and counts should be the same across all clusters.  The framework does not enforce or validate this, but should be configured equivalently on all clusters.
```
Comment (joel-hamill) [inline docs/docs/multi-dc.md]: Change to: "Node sizes and counts must be the same across all clusters.  The framework does not enforce or validate this, but this must be configured equivalently on all clusters."

File: docs/docs/multi-dc.md [hunk 16]
```diff
@@ -0,0 +1,176 @@
...
+```
+
+### Other instances of Cassandra on Mesos
+
+Cassandra on Mesos is informed of other instances of Cassandra on Mesos (running in another Mesos cluster) by
+environment variables of the format `CASSANDRA_EXTERNAL_DC_<dcName>=<cassandraOnMesosUrl>`.
+
+## Example on Mesos
+
+Imagine three Data Centers dc0, dc1, dc2 running three different Mesos clusters.
```
Comment (joel-hamill) [inline docs/docs/multi-dc.md]: Change to: "In this example, three Data Centers dc0..."

File: docs/docs/multi-dc.md (hunk 3 above)
Comment (joel-hamill) [inline docs/docs/multi-dc.md]: Change to: "In this example, three datacenters..."

File: docs/docs/multi-dc.md (hunk 7 above)
Comment (joel-hamill) [inline docs/docs/multi-dc.md]: Change to: "After the seeds have been allocated, you must restart all of the nodes that have already been started. "

**Comment by BenWhitehead** [comment]
Thanks for the review @ssk2 and @joel-hamill.

Updates in place. I'll squash before merging them.

**Comment by mattpennathe3rd** [comment]
@BenWhitehead @joel-hamill Made my own pass through this and made a number of changes. Please let me know how this looks and if my method of proposing/committing these changes was improper.

**Comment by ssk2** [comment]
@mattpennathe3rd perfectly acceptable method of reviewing :)

**Comment by BenWhitehead** [comment]
@mattpennathe3rd Thanks for the feedback.  I've made only a couple modifications to your proposal specifically to make sure that when we're talking about Cassandra on Mesos we use the full name so that people don't think it's about cassandra itself.

**Comment by BenWhitehead** [comment]
going to rebase then merge.  Thanks @ssk2 @joel-hamill @mattpennathe3rd
